experiments/mnist_lr_brittleness.py

```diff
@@ -0,0 +1,140 @@
+    )
+    
+    # test samples for attribution
+    test_loader = torch.utils.data.DataLoader(
+        dataset,
+        batch_size=500,
+        sampler=SubsetSampler(range(1000)),
+    )
+
+    model = train_mnist_lr(train_loader_full)
+    model.cuda()
+    model.eval()
+
+    
+    def f(params, data_target_pair):
+        image, label = data_target_pair
+        loss = nn.CrossEntropyLoss()
+        yhat = torch.func.functional_call(model, params, image)
+        return loss(yhat, label.long())
+
+    task = AttributionTask(target_func=f,
+                           model=model,
+                           checkpoints=model.state_dict())
+    attributor = ATTRIBUTOR_MAP[args.method](
+        task=task,
+        device=args.device,
+    )
+
+    attributor.cache(train_loader_full)
+    torch.cuda.reset_peak_memory_stats("cuda")
+    with torch.no_grad():
+        score = attributor.attribute(train_loader, test_loader)
+
+    model = train_mnist_lr(train_loader)
+    model.cuda()
+    model.eval()
+    correct_x = None
+    correct_label = None
+    correct_index = None
+
+    # Find a correct predicted data
+    for test_batch in test_loader:
+        for i in range(len(test_batch[0])):
+            i = i
+            x = test_batch[0][i].unsqueeze(0).cuda()
+            label = test_batch[1][i].cuda()
+            output = model(x)
+            pred = output.argmax(dim=1, keepdim=True)
+
+            if pred.item() == label.item():
+                print(f"Found a correctly predicted test sample with correct index is: {i}")
+                print(f"The label: {label.item()}, the prediction: {pred.item()}")
+                correct_x = x.unsqueeze(0)
+                correct_label = label.unsqueeze(0)
+                correct_index = i
+                break
+
+        if correct_x is not None:
+            break
+    
+    # Example test sample
+    test_dataset = TensorDataset(correct_x, correct_label)
+    test_loader = DataLoader(test_dataset, batch_size=1, shuffle=True)
+    score = score[:, correct_index]
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+    # Compute brittleness
+    start_time = time.time()
+
+    # Evaluation 
+    def eval_func(model_output):
+        probabilities = torch.softmax(model_output, dim=1)
+        predicted_class = torch.argmax(probabilities, dim=1)
+        return predicted_class
+
+    smallest_k = brittleness(
+        train_loader=train_loader,
+        test_loader=test_loader,
+        scores=score,
+        train_func=lambda loader: train_mnist_lr(loader),
+        eval_func=eval_func,
+        device=device,
+        search_space=[20,40,60,80,100,120,140,180],
+    )
+    if smallest_k is not None:
+        print(f"The number of removal that can make it flip: {smallest_k}")
+    else:
+        print("No k found that causes a flip.")
+    end_time = time.time()
+    print("Total time for brittleness test: ", end_time - start_time)
```

Comment:
maybe paste some result here? so that it's easier to see if the implementation give reasonable result.

Reply:
![image](https://github.com/user-attachments/assets/b25c5912-fce7-4410-bcb0-a964b1fed5a1)
Here is the result of doing brittleness test on the mnist+mlp as the benchmark. The mlp model is trained on 500 size mnist data. It shows 4 different TDA method: IF, RPS, TRAK, TracIn.